vertx-web/src/main/java/io/vertx/ext/web/Router.java

```diff
@@ -395,4 +396,13 @@ default void accept(HttpServerRequest request) {
    */
   void handleFailure(RoutingContext context);
 
+  /**
+   * When a Router routes are changed this handler is notified.
+   * This is useful for routes that depend on the state of the router.
+   *
+   * @param handler a notification handler that will receive this router as argument
+   * @return a reference to this, so the API can be used fluently
+   */
+  @Fluent
+  Router modifiedHandler(Handler<Router> handler);
```

Comment:
I'm wondering, can we apply this one to https://github.com/vert-x3/vertx-web/pull/1376 ? So on router changes we can merge the changes to copied routes